File: buzzad-benefit-push/app/src/main/java/com/buzzvil/buzzad/benefit/notiplussample/CustomNotificationWorker.java
package com.buzzvil.buzzad.benefit.notiplussample;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.WorkerParameters;

import com.buzzvil.buzzad.benefit.presentation.feed.FeedConfig;
import com.buzzvil.buzzad.benefit.presentation.notification.NotificationConfig;
import com.buzzvil.buzzad.benefit.presentation.notification.NotificationWorker;

public class CustomNotificationWorker extends NotificationWorker {
    public CustomNotificationWorker(@NonNull Context context,
                                    @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @Override
    @NonNull
    public NotificationConfig getNotificationConfig() {
        return new NotificationConfig.Builder(App.UNIT_ID_FEED)
                .build();
    }
}

File: archive/demo/app/src/main/java/com/buzzvil/buzzad/demo/customized/CustomNotificationWorker.kt
package com.buzzvil.buzzad.demo.customized

import android.content.Context
import androidx.work.WorkerParameters
import com.buzzvil.buzzad.benefit.presentation.notification.NotificationConfig
import com.buzzvil.buzzad.benefit.presentation.notification.NotificationWorker

class CustomNotificationWorker(
    context: Context,
    workerParams: WorkerParameters
) : NotificationWorker(context, workerParams) {

    override val notificationConfig: NotificationConfig
        get() = NotificationConfig.Builder()
            .notificationId(1234)
            .build()
}
